### PR TITLE
Combobox design fixes

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://github.com/adobe/react-spectrum/issues/1444

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/fieldlabel/index.css [hunk 1]
```diff
@@ -71,6 +71,18 @@ governing permissions and limitations under the License.
...
+      padding-bottom: var(--spectrum-fieldlabel-quiet-padding-bottom);
+      padding-top: var(--spectrum-fieldlabel-quiet-padding-top);
+      line-height: var(--spectrum-fieldlabel-quiet-text-line-height);
+    }
+
+    .spectrum-Field-fieldInput--quiet {
+      padding-top: var(--spectrum-textfield-quiet-padding-top);
+      padding-bottom: var(--spectrum-textfield-quiet-padding-bottom);
+      line-height: var(--spectrum-textfield-quiet-text-line-height);
+    }
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/fieldlabel/index.css:85]: Values mimic what I saw in the XD file, but feels a bit off still. Doesn't feel like it matches the 20px mentioned in the ticket...

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 2]
```diff
@@ -120,6 +139,17 @@ governing permissions and limitations under the License.
...
     }
+
+    &:focus-ring {
+      .spectrum-InputGroup-input {
+        box-shadow: 0 1px 0 var(--spectrum-textfield-quiet-border-color-key-focus);
+      }
+
+      .spectrum-FieldButton {
+        box-shadow: 0 1px 0 var(--spectrum-textfield-quiet-border-color-key-focus);
+        border-color: var(--spectrum-textfield-quiet-border-color-key-focus);
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:150]: As per design feedback, only show the box-shadow if the combobox is keyboard focused.

File: packages/@adobe/spectrum-css-temp/components/inputgroup/index.css [hunk 3]
```diff
@@ -23,7 +23,7 @@ governing permissions and limitations under the License.
   --spectrum-datepicker-range-dash-padding-top: 0;
   --spectrum-datepicker-range-dash-line-height: calc(var(--spectrum-textfield-height) - var(--spectrum-global-dimension-size-100));
 
-  --spectrum-combobox-button-width: calc(var(--spectrum-global-dimension-size-400) + var(--spectrum-global-dimension-size-25));
+  --spectrum-combobox-button-width: var(--spectrum-global-dimension-size-400);
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/index.css:26]: Make button 32x32

File: packages/@adobe/spectrum-css-temp/components/inputgroup/index.css [hunk 4]
```diff
@@ -95,6 +95,12 @@ governing permissions and limitations under the License.
   }
 }
 
+.spectrum-InputGroup-popover--quiet {
+  /* Define this var so it can be read from JS */
+  --spectrum-dropdown-quiet-offset: calc(var(--spectrum-dropdown-quiet-popover-offset-x) + var(--spectrum-popover-border-size));
+  margin-inline-end: calc(var(--spectrum-dropdown-quiet-offset) * -1);
+}
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/index.css:102]: Offsets the quiet combobox menu so menu text lines up with combobox test

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 5]
```diff
@@ -34,6 +34,33 @@ governing permissions and limitations under the License.
...
+        &:not(.is-focused){
+          .spectrum-FieldButton {
+            border-color: var(--spectrum-global-color-gray-500);
+          }
+        }
+
+        .spectrum-FieldButton {
+          &:hover .spectrum-Dropdown-chevron {
+            color: var(--spectrum-global-color-gray-900);
+          }
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:59]: For the `The button should retain the "down" style while the menu is open` design feedback

File: packages/@adobe/spectrum-css-temp/components/textfield/skin.css [hunk 6]
```diff
@@ -115,7 +115,6 @@ governing permissions and limitations under the License.
     &:focus,
     &.is-focused {
       border-color: var(--spectrum-textfield-quiet-border-color-key-focus);
-      box-shadow: 0 1px 0 var(--spectrum-textfield-quiet-border-color-key-focus);
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/textfield/skin.css:118]: Applied the box-shadow feedback to textfield as per textfield 5.3 beta XD file

File: packages/@react-spectrum/combobox/src/ComboBox.tsx [hunk 7]
```diff
@@ -118,14 +122,39 @@ const ComboBoxBase = React.forwardRef(function ComboBoxBase<T extends object>(pr
...
+    labelClassName = classNames(
+      labelStyles,
+      'spectrum-FieldLabel--quiet'
+    );
+
+    inputClassName = classNames(
+      labelStyles,
+      'spectrum-Field-fieldInput--quiet'
+    );
+  }
```
Comment (LFDanLu) [inline packages/@react-spectrum/combobox/src/ComboBox.tsx:141]: For the 28px -> 20px spacing between label and bottom of combobox input text design feedback for quiet comboboxes

**Comment by dannify** [comment]
Invalid quiet still seems to have a 2px focus "ring" instead of 1px when you open the menu with a mouse.

image

**Review by dannify — APPROVED** [review]

**Pull request merged**